pkg/kine/drivers/sqlite/sqlite.go

```diff
@@ -121,20 +121,15 @@ func NewVariant(ctx context.Context, driverName, dataSourceName string) (server.
 // changes are rolled back if an error occurs.
 func setup(ctx context.Context, db *sql.DB) error {
 	// Optimistically ask for the user_version without starting a transaction
-	var schemaVersion int32
+	var currentSchemaVersion SchemaVersion
 
 	row := db.QueryRowContext(ctx, `PRAGMA user_version`)
-	if err := row.Scan(&schemaVersion); err != nil {
+	if err := row.Scan(&currentSchemaVersion); err != nil {
 		return err
 	}
-	currentMajor, currentMinor := getMajorMinorVersion(schemaVersion)
 
-	if currentMajor == databaseSchemaMajorVersion && currentMinor == databaseSchemaMinorVersion {
-		return nil
-	} else if currentMajor > 0 && currentMajor < databaseSchemaMajorVersion {
-		return errors.Errorf("can not update major versions: %d to %d", currentMajor, databaseSchemaMajorVersion)
-	} else if currentMajor > databaseSchemaMajorVersion {
-		return errors.Errorf("can not rollback to earlier major version: %d to %d", currentMajor, databaseSchemaMajorVersion)
+	if canMigrate, err := currentSchemaVersion.CanMigrate(databaseSchemaVersion); !canMigrate {
```

Comment:
I'd prefer to have a more explicit logic here. Consider that in this place 2 things are decided:
 - if we should run migration;
 - if we can work with the current schema (for example, the current schema is a newer version, but it is compatible). 
 
 So, I would prefer something on the lines of:
 
 ```go
 if err := databaseSchemaVersion.CompatibleWith(currentSchemaVersion); err != nil {
   return err
 }
 if currentSchemaVersion >= databaseSchemaVersion {
   // Already to the latest version.
   return nil
 }
 ```
 
 This would change the `CanMigrate` method on the `SchemaVersion` which is very specific to our migration logic (which might change) to a much more generic `CompatibleWith` which is a generic method over semantic versioning.

Reply:
Yes, that seems like a better way!